### PR TITLE
INFRA-13873 Replace libinjection dependency

### PR DESCRIPTION
**Pull request by jimmybigcommerce**
The current `libinjection` library is unmaintained and incompatible with Python 3. This PR switches to the https://github.com/libinjection/libinjection library which is still maintained. This required some changes to the Makefile to get it to build properly.

As part of this, I needed to fix a number of tests that were broken as they were based on the premise that lua hash tables always return items in the same order, which they do not.

## Proof

Ran tests - output attached below

[lua-resty-waf-test-output.txt](https://github.com/bigcommerce/lua-resty-waf/files/15269690/lua-resty-waf-test-output.txt)

### CODE REVIEW

File: lib/resty/htmlentities.lua [hunk 1]
```diff
@@ -0,0 +1,64 @@
+local ffi = require('ffi')
```
Comment (rayward) [inline lib/resty/htmlentities.lua]: Is this file meant to be committed in, rather than coming from the submodule?

Comment (jimmybigcommerce) [reply lib/resty/htmlentities.lua]: Yep you are right. Have removed and added to .gitignore

File: lib/resty/waf.lua [hunk 2]
```diff
@@ -209,7 +209,7 @@ local function _do_transform(self, collection, transform)
 				return collection -- dont transform if the collection was nil, i.e. a specific arg key dne
 			end
 
-			--_LOG_"doing transform of type " .. transform .. " on collection value " .. tostring(collection)
+			if self._debug == true then ngx.log(self._debug_log_level, '[', self.transaction_id, '] ', "doing transform of type " .. transform .. " on collection value " .. tostring(collection)) end
```
Comment (rayward) [inline lib/resty/waf.lua]: Did the makefile make all these changes here with the debug stuff?

Comment (jimmybigcommerce) [reply lib/resty/waf.lua]: Yeah

Comment (jimmybigcommerce) [reply lib/resty/waf.lua]: I wasn't certain whether I should commit it or not

Comment (rayward) [reply lib/resty/waf.lua]: I don't think we'd ever enable debug logging on the waf, unless you think we'd want that capability for testing with this new build? 

It'd really just be a micro-optimisation to skip checking the `_debug` property.

Thoughts?

Comment (jimmybigcommerce) [reply lib/resty/waf.lua]: :+1: Have removed these changes

**Review by rayward — APPROVED** [review]

**Pull request merged**